### PR TITLE
Hw3 added

### PR DESCRIPTION
**Pull request by sarvesh2910**
This pr is about hw3

### CODE REVIEW

File: hw3/01-routing.js [hunk 1]
```diff
@@ -13,37 +12,84 @@ const port = process.env.PORT || 5001;
...
+      res.writeHead(302, { Location: "/redirected" });
+      res.end();
+      break;
+    case "/redirected":
+      res.setHeader("Content-Type", "text/html");
+      res.write(`Cached!`);
+      res.end();
+      break;
+    case "/cache":
+      res.setHeader("Content-Type", "text/html");
```
Comment (ethanSE) [inline hw3/01-routing.js:71]: I believe that by "in plain text" she meant setting the context type to "plaintext"

edit: I'm wrong it is text/plain as you pointed out on my PR

File: hw3/02-url.js [hunk 2]
```diff
@@ -1,38 +1,47 @@
...
-
-    res.writeHead(200, { 'Content-Type': 'text/html' });
+  res.writeHead(200, { "Content-Type": "text/html" });
+  if (req.url === "/") {
+    const routeResults = getRoutes(routes);
     res.write(`<h1>Exercise 02</h1>`);
-
     res.write(`<ul> ${routeResults} </ul>`);
+  } else {
+    const queryObject = urlParser.parse(req.url, true).query;
```
Comment (ethanSE) [inline hw3/02-url.js:41]: Interesting. I used URLSearchParams and haven't dug too deeply into doing it this way

**Review by ethanSE — APPROVED** [review]

**Comment by prachi-kashyap** [comment]
Your code seems very well written. Just a small observation - You could have created a new file for getRoutes() and used it in both the routing.js and url.js files. 
I don't know if it is a best practice or not. Maybe you can explain why it should be used or not if you know the reason.

**Review by prachi-kashyap — APPROVED** [review]

File: hw3/01-routing.js (hunk 1 above)
Comment (sarvesh2910) [reply hw3/01-routing.js:71]: No worries. 
when I used plain text, it really was plain text on black background, which I thought I could improve, hence used text HTML.

**Comment by sarvesh2910** [comment]
Yup. @prachi-kashyap. I could have done it. I saw it in @ethanSE PR.

**Pull request merged**